src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/Repo.scala

```diff
@@ -79,6 +81,89 @@ class Repo[F[_]: Monad](agg: Agg[F], clock: Clock, toIdentifier: ResId => String
       implicit identity: Identity): EitherT[F, Rejection, Resource] =
     evaluate(id, AddTag(id, rev, targetRev, tag, clock.instant(), identity))
 
+  /**
+    * Adds an attachment to a resource.
+    *
+    * @param id     the id of the resource
+    * @param rev    the last known revision of the resource
+    * @param attach the attachment description metadata
+    * @param source the source of the attachment
+    * @tparam In the storage input type
+    * @return either a rejection or the new resource representation in the F context
+    */
+  def attach[In](id: ResId, rev: Long, attach: BinaryDescription, source: In)(
+      implicit identity: Identity,
+      store: AttachmentStore[F, In, _]): EitherT[F, Rejection, Resource] =
+    get(id).toRight(NotFound(id.ref)).flatMap { res =>
+      if (res.deprecated) EitherT.leftT(IsDeprecated(id.ref))
```

Comment:
couldn't we use `agg.checkEval` here?

Reply:
Yes, but in practice it becomes more ugly because we have to create a dummy command and event for that. I think this is a neater way.

Reply:
We would have to create a dummy instance of a command, but no events:

https://github.com/BlueBrain/nexus-sourcing/blob/b93baefd83d5bf053fc83cdd16d32055a927ac55/modules/core/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/Aggregate.scala#L39

But I see your point, it's fine to keep it like this.